### ISSUE TITLE
Server Crashes 

### ISSUE THREAD

**Issue by BaraHashesh** [TooTallNate/Java-WebSocket#795]
**Unintended crashes**
In some cases the Server completely crashes if the **onMessage()** methods take too long to finish
the replay/processing process, this may happen when transferring (download from server/upload to server) large files/folders at low speeds.

**To Reproduce**
1. Define the onMessage() methods for the server
2. Make the executed code such as it takes too long to finish +2 minutes (sleep method for example). 

**Environment(please complete the following information):**
 - Version used: 1.3.9
 - Java version: JDK8
 - Operating System and version: windows 10 

**Note**

A quick fix to this is wrapping the code with a wrapping thread.

**Comment by marci4** [comment]
Hello @BaraHashesh,

please use the full template and include a project with easy steps to reproduce!

Best regards,
marci4

**Comment by BaraHashesh** [comment]
There was no need to include the full template, furthermore i think i made the problem and how to recreate it quite simple, however since it's unclear as you say here's a code to recreate it.

        import java.io.BufferedReader;
        import java.io.IOException;
        import java.io.InputStreamReader;
        import java.net.InetSocketAddress;
        import java.net.UnknownHostException;
        import java.nio.ByteBuffer;

        import org.java_websocket.WebSocket;
        import org.java_websocket.WebSocketImpl;
        import org.java_websocket.framing.Framedata;
        import org.java_websocket.handshake.ClientHandshake;
        import org.java_websocket.server.WebSocketServer;

        /**
         * A simple WebSocketServer implementation. Keeps track of a "chatroom".
         */
        public class ChatServer extends WebSocketServer {

            public ChatServer( int port ) throws UnknownHostException {
                super( new InetSocketAddress( port ) );
            }

            public ChatServer( InetSocketAddress address ) {
                super( address );
            }

            @Override
            public void onOpen( WebSocket conn, ClientHandshake handshake ) {
                conn.send("Welcome to the server!"); //This method sends a message to the new client
                broadcast( "new connection: " + handshake.getResourceDescriptor() ); //This method sends a 
                message to all clients connected
                System.out.println( conn.getRemoteSocketAddress().getAddress().getHostAddress() + " 
                entered the room!" );
            }

            @Override
            public void onClose( WebSocket conn, int code, String reason, boolean remote ) {
                try {
                    Thread.Sleep(1000*60*2);
                } catch (Exception e) {
                    //TODO: handle exception
                }
                broadcast( conn + " has left the room!" );
                System.out.println( conn + " has left the room!" );
            }

            @Override
            public void onMessage( WebSocket conn, String message ) {
                broadcast( message );
                System.out.println( conn + ": " + message );
            }
            @Override
            public void onMessage( WebSocket conn, ByteBuffer message ) {
                broadcast( message.array() );
                System.out.println( conn + ": " + message );
            }


            public static void main( String[] args ) throws InterruptedException , IOException {
                int port = 8887; // 843 flash policy port
                try {
                    port = Integer.parseInt( args[ 0 ] );
                } catch ( Exception ex ) {
                }
                ChatServer s = new ChatServer( port );
                s.start();
                System.out.println( "ChatServer started on port: " + s.getPort() );

                BufferedReader sysin = new BufferedReader( new InputStreamReader( System.in ) );
                while ( true ) {
                    String in = sysin.readLine();
                    s.broadcast( in );
                    if( in.equals( "exit" ) ) {
                        s.stop(1000);
                        break;
                    }
                }
            }
            @Override
            public void onError( WebSocket conn, Exception ex ) {
                ex.printStackTrace();
                if( conn != null ) {
                    // some errors like port binding failed may not be assignable to a specific websocket
                }
            }

            @Override
            public void onStart() {
                System.out.println("Server started!");
                setConnectionLostTimeout(0);
                setConnectionLostTimeout(100);
            }

        }

**Comment by marci4** [comment]
Hello @BaraHashesh,

the reason why I require the template is simple because it reduces the change for a misunderstanding and also helps me use my limit time available more effective.

Like for example with this issue here.
I cannot reproduce it with the current master branch and I also do not think it is somehow fixed already.
That is why I need clean steps to reproduce and also a link to an example application for easier reproduction. Since I followed your first report, I used sleep in onMessage and not onClose as you used it in your answer.

It is also not described what you mean by crash etc etc.

Best regards,
marci4
